pkg/webhooks/admission/pods/validate/admit_pod.go

```diff
@@ -136,12 +143,25 @@ func validatePod(pod *v1.Pod, reviewResponse *admissionv1.AdmissionResponse) str
 }
 
 func checkPG(pod *v1.Pod, pgName string, isVCJob bool) error {
-	_, err := config.VolcanoClient.SchedulingV1beta1().PodGroups(pod.Namespace).Get(context.TODO(), pgName, metav1.GetOptions{})
+	pgObj, err := config.VolcanoClient.SchedulingV1beta1().PodGroups(pod.Namespace).Get(context.TODO(), pgName, metav1.GetOptions{})
 	if err != nil {
 		if isVCJob || (!isVCJob && !apierrors.IsNotFound(err)) {
 			return fmt.Errorf("failed to get PodGroup for pod <%s/%s>: %v", pod.Namespace, pod.Name, err)
 		}
 		return nil
+	} else if errQueue := checkQueueState(pgObj.Spec.Queue); errQueue != nil {
```

Comment:
why check queue state in `checkPG` func? l think maybe they should be separated

Reply:
The only reason to check inside `checkPG` is to get `queueName` from `pgObj`.
I can move it outside `checkPG`, but need to make one more call to `config.VolcanoClient.SchedulingV1beta1().PodGroups(pod.Namespace).Get(context.TODO(), pgName, metav1.GetOptions{})`

That is fine?